### DISCUSSION TITLE
Can you add tag descriptions through an APIRouter?

### DISCUSSION THREAD

**Discussion by p3l** [fastapi/fastapi#5309] [Questions] [answered]
If I set tags trough the FastAPI() object I can set tag descriptions using the openapi_tags in this way:

> tags_metadata = [{'name': 'foo', 'description': 'this is a foo'}]
> app = FastAPI(openapi_tags=tags_metadata)

But how do I do the same if I am adding tags to an APIRouter?

> router = APIRouter(tags=['foo'])

**Comment by luebke-dev** [comment]
You do it when you include the API Router


app.include_router(router, prefix="/test", tags=["mytag"])

**Reply by p3l** [reply]
Well, your suggestion has the same problem as the one I am trying to solve: it can only set the tags, not the tag description.
If you look at my first example you will see that the openapi_tags attribute is a list of dictionaries, not a list of strings.

**Reply by yezz123** [reply]
For the tags, you could use this package https://github.com/yezz123/fastapi-tag used to enhance the metadata of your Project and document your API with more information.

**Comment by moreno-p** [comment, marked as answer]
Yes, you can: 

from fastapi import FastAPI

tags_metadata = [
    {
        "name": "users",
        "description": "Operations with users. The **login** logic is also here.",
    },
    {
        "name": "items",
        "description": "Manage items. So _fancy_ they have their own docs.",
        "externalDocs": {
            "description": "Items external docs",
            "url": "https://fastapi.tiangolo.com/",
        },
    },
]

app = FastAPI(openapi_tags=tags_metadata)


@app.get("/users/", tags=["users"])
async def get_users():
    return [{"name": "Harry"}, {"name": "Ron"}]


From the official doc: https://fastapi.tiangolo.com/tutorial/metadata/

**Reply by gabriella-speariett** [reply]
@p3l Sorry how does this answer your question? The description is still not on the router?
